### DISCUSSION TITLE
TestHarness does not require python3

### DISCUSSION THREAD

**Discussion by hzclarksm** [idaholab/moose#24048] [Q&A Getting Started] [answered]
PythonUnitTest.getCommand() does explicitly use python3.  In my environment "python" is by default python2.   This cases several tests to fail which missing libraries and and/or syntax errors.  Here is a simple fix.

```diff
diff --git a/python/TestHarness/testers/PythonUnitTest.py b/python/TestHarness/testers/PythonUnitTest.py
index 1afb363c31..1a77cd6e92 100644
--- a/python/TestHarness/testers/PythonUnitTest.py
+++ b/python/TestHarness/testers/PythonUnitTest.py
@@ -44,6 +44,6 @@ class PythonUnitTest(RunApp):
         if self.specs["separate"]:
             cmd = os.path.join(self.specs['moose_dir'], 'scripts', 'separate_unittests.py') + ' -f ' + test_case + use_buffer
         else:
-            cmd = "python -m unittest" + use_buffer + "-v " + test_case
+            cmd = "python3 -m unittest" + use_buffer + "-v " + test_case
 
         return cmd  + ' '.join(self.specs['cli_args'])
```

**Comment by GiudGiud** [comment]
@milljm can we do that?

Is python3 (in addition to python) provided on every platform we care about?

**Reply by milljm** [reply]
Bringing in @dschwen, because his voice on this is like mine :)

I say we change all our Python shebangs to `#!/usr/bin/env python3` If not already. I thought I had sneaked this in already at some point.

**Reply by milljm** [reply, marked as answer]
but yes, I agree with the change. When calling `python` we should call `python3`

**Reply by hzclarksm** [reply]
The change to Python shebangs does seem to already be in place.  That of course has no effect when running python commands as built in PythonUnitTest.py.
